### PR TITLE
feat(plasma-tokens): Add typography generator

### PR DESCRIPTION
**Pull request by neretin-trike**
:baby_chick: Download canary assets:

[colors--canary.198.3362763601.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.198.3362763601.xml)  
[color_sbermarket_ios-swift--canary.198.3362763601.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.198.3362763601.swift)  
[color_sbermarket_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.198.3362763601.kt)  
[color_sbermarket_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.198.3362763601.ts)  
[color_sberprime_ios-swift--canary.198.3362763601.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.198.3362763601.swift)  
[color_sberprime_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.198.3362763601.kt)  
[color_sberprime_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.198.3362763601.ts)  
[PlasmaTokensColor--canary.198.3362763601.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.198.3362763601.swift)  
[typo_mage_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.198.3362763601.kt)  
[typo_mage_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.198.3362763601.ts)  
[typo_plasma_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.198.3362763601.kt)  
[typo_plasma_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.198.3362763601.ts)  
[typo_ruler_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.198.3362763601.kt)  
[typo_ruler_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.198.3362763601.ts)  
[typo_sage_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.198.3362763601.kt)  
[typo_sage_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.198.3362763601.ts)  
[typo_sbermarket_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.198.3362763601.kt)  
[typo_sbermarket_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.198.3362763601.ts)  
[typo_soulmate_kotlin--canary.198.3362763601.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.198.3362763601.kt)  
[typo_soulmate_react-native--canary.198.3362763601.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.198.3362763601.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.109.0-canary.198.3362763601.0
  npm install @salutejs/plasma-core@1.81.0-canary.198.3362763601.0
  npm install @salutejs/plasma-icons@1.106.0-canary.198.3362763601.0
  npm install @salutejs/plasma-temple@1.108.0-canary.198.3362763601.0
  npm install @salutejs/plasma-tokens@1.28.0-canary.198.3362763601.0
  npm install @salutejs/plasma-typo@0.28.0-canary.198.3362763601.0
  npm install @salutejs/plasma-ui@1.142.0-canary.198.3362763601.0
  npm install @salutejs/plasma-web@1.140.0-canary.198.3362763601.0
  npm install @salutejs/plasma-cy-utils@0.34.0-canary.198.3362763601.0
  npm install @salutejs/plasma-sb-utils@0.79.0-canary.198.3362763601.0
  npm install @salutejs/plasma-tokens-android@2.35.0-canary.198.3362763601.0
  npm install @salutejs/plasma-tokens-ios-swift@2.35.0-canary.198.3362763601.0
  # or 
  yarn add @salutejs/plasma-b2c@1.109.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-core@1.81.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-icons@1.106.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-temple@1.108.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-tokens@1.28.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-typo@0.28.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-ui@1.142.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-web@1.140.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-cy-utils@0.34.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-sb-utils@0.79.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-tokens-android@2.35.0-canary.198.3362763601.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.35.0-canary.198.3362763601.0
  ```
</details>

### CODE REVIEW

File: packages/plasma-tokens/config.ts [hunk 1]
```diff
@@ -11,23 +11,38 @@ import {
...
+// const platforms = ['kotlin', 'react-native', 'react-native-typo', 'ios-swift'];
 
-const tokensPath = `properties/**/brands`;
+// const tokensColorPath = `properties/color/brands`;
+const tokensTypoPath = `properties/typo/`;
 
-const fileNames = glob.sync(path.join(tokensPath, '*')).map((filePath) => path.parse(filePath).name);
+const fileNames = ['sbermarket']; //glob.sync(path.join(tokensTypoPath, '*')).map((filePath) => path.parse(filePath).name);
 
 const getStyleDictionaryConfig = (brand: string) => ({
```
Comment (Yakutoc) [inline packages/plasma-tokens/config.ts:25]: Подскажи у нас есть `type` для описание того, что возвращает эта функция?

File: packages/plasma-typo/src/helpers.ts [hunk 2]
```diff
@@ -37,17 +37,37 @@ export const prepareCompatibleTypo = (typoMap: TypoMap) =>
...
-        return {
-            ...acc,
-            [key]: {
-                ...acc[key as TypoPropName],
-                ...value,
-            },
-        };
-    }, obj1);
-};
+export const mergeTypoProps = (...obj: TypoProps[]): TypoProps =>
```
Comment (Yakutoc) [inline packages/plasma-typo/src/helpers.ts:40]: [Minor]: А если использовать https://lodash.com/docs/#merge для таких вещей?

Comment (neretin-trike) [reply packages/plasma-typo/src/helpers.ts]: не, обычно мы не тянем дополнительных зависимостей, если нужно что-то небольшое написать своими руками

File: packages/plasma-tokens/config.ts (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-tokens/config.ts]: ага, добавил

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-198/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-198/
  ui storybook: http://plasma.sberdevices.ru/pr-198/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-198/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-198/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-198/temple-storybook/

File: packages/plasma-tokens/lib/generator/formatters/typoReactNative.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    constTypoStyles: string,
+    exportTypoStylesContent: string,
+    baseFontFamily: string,
+    additionalFontFamily: string,
+) => {
+    const isSameFont = baseFontFamily === additionalFontFamily;
+
+    const header = `/* eslint-disable @typescript-eslint/no-magic-numbers */
+import type { TextStyle } from "react-native";
+import styled from "styled-components/native";
```
Comment (Yeti-or) [inline packages/plasma-tokens/lib/generator/formatters/typoReactNative.ts:18]: Может что то такое для улучшения понимания?

https://github.com/dmnd/dedent

File: packages/plasma-tokens/package.json [hunk 4]
```diff
@@ -46,20 +46,18 @@
...
         "@salutejs/plasma-tokens-utils": "0.13.0",
+        "@salutejs/plasma-typo": "0.26.0",
         "@types/inquirer": "8.2.1",
         "@types/node": "16.7.13",
         "@types/param-case": "1.1.2",
         "inquirer": "8.2.0",
         "jsdom": "17.0.0",
         "jsdom-global": "3.0.2",
         "style-dictionary": "3.7.1",
+        "styled-components": "5.3.6",
```
Comment (Yeti-or) [inline packages/plasma-tokens/package.json:57]: Я так и не понял зачем они нужны тут

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-tokens/lib/generator/formatters/typoReactNative.ts (hunk 3 above)
Comment (Yeti-or) [reply packages/plasma-tokens/lib/generator/formatters/typoReactNative.ts:18]: https://github.com/zspecza/common-tags

File: packages/plasma-tokens/package.json (hunk 4 above)
Comment (neretin-trike) [reply packages/plasma-tokens/package.json:57]: удалил 👌

File: packages/plasma-tokens/lib/generator/formatters/typoReactNative.ts (hunk 3 above)
Comment (neretin-trike) [reply packages/plasma-tokens/lib/generator/formatters/typoReactNative.ts:18]: эти либы тоже давно уже не обновлялись, поэтому я бы не стал к нам затаскивать. 
Можно подумать над своим решением, попробую его сделать при реализации шаблонов для swfit'а

Comment (Yeti-or) [reply packages/plasma-tokens/lib/generator/formatters/typoReactNative.ts:18]: А может они хорошо написаны и не требуют доработок ? ;)

**Comment by Yeti-or** [comment]
:shipit:

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-198/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-198/
  ui storybook: http://plasma.sberdevices.ru/pr-198/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-198/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-198/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-198/temple-storybook/

**Pull request merged**